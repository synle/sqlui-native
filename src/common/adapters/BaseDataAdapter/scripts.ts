import { getDialectType } from 'src/common/adapters/DataScriptFactory';
import IDataScript from 'src/common/adapters/IDataScript';
import { SqlAction, SqluiCore } from 'typings';

export function getDivider(): SqlAction.Output {
  return {
    label: 'divider',
    skipGuide: true,
  };
}

export default abstract class BaseDataScript implements IDataScript {
  dialects: string[] = [];

  isDialectSupported(dialect?: string) {
    return !!dialect && this.dialects.indexOf(dialect) >= 0;
  }

  getConnectionFormInputs() {
    return [
      ['username', 'Username'],
      ['password', 'Password'],
      ['host', 'Host'],
      ['port', 'Port', 'optional'],
    ];
  }

  getIsTableIdRequiredForQuery() {
    return false;
  }

  getSyntaxMode() {
    return 'sql';
  }

  supportMigration() {
    return false;
  }

  supportCreateRecordForm() {
    return false;
  }

  supportEditRecordForm() {
    return false;
  }

  // dialect definitions
  getDialectType(dialect?: SqluiCore.Dialect) {
    // attempt to return the first item in the dialects / schemes
    if (dialect && this.dialects.indexOf(dialect) >= 0) {
      return dialect as SqluiCore.Dialect;
    }

    return undefined;
  }

  getDialectName(dialect?: SqluiCore.Dialect): string {
    // capitalize the first letter
    return (dialect || '').replace(/^\w/, (c) => c.toUpperCase());
  }

  getDialectIcon(dialect?: SqluiCore.Dialect): string {
    return `${process.env.PUBLIC_URL}/assets/${dialect}.png`;
  }

  // sample data script
  getTableScripts(): SqlAction.TableActionScriptGenerator[] {
    return [];
  }

  getDatabaseScripts(): SqlAction.DatabaseActionScriptGenerator[] {
    return [];
  }

  getConnectionScripts(): SqlAction.ConnectionActionScriptGenerator[] {
    return [];
  }

  getSampleConnectionString(dialect?: SqluiCore.Dialect) {
    return '';
  }

  getSampleSelectQuery(actionInput: SqlAction.TableInput): SqlAction.Output | undefined {
    return undefined;
  }

  // sample code snippet
  getCodeSnippet(
    connection: SqluiCore.ConnectionProps,
    query: SqluiCore.ConnectionQuery,
    language: SqluiCore.LanguageMode,
  ) {
    switch (language) {
      case 'javascript':
        // TODO: implement me
        return '';
      case 'python':
        // TODO: implement me
        return '';
      case 'java':
        // TODO: implement me
        return '';
      default:
        return '';
    }
  }
}
