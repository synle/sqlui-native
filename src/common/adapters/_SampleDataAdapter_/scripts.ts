import BaseDataScript, { getDivider } from 'src/common/adapters/BaseDataAdapter/scripts';
import { SqlAction, SqluiCore } from 'typings';

const formatter = 'js';

export function getSelectAllColumns(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Select All Columns`;

  return {
    label,
    formatter,
    query: `// TODO: implement me`,
  };
}

export function getCreateDatabase(input: SqlAction.DatabaseInput): SqlAction.Output | undefined {
  const label = `Create Database`;

  return {
    label,
    formatter,
    query: `
      // TODO: implement me
    `,
  };
}

export function getDropDatabase(input: SqlAction.DatabaseInput): SqlAction.Output | undefined {
  const label = `Drop Database`;

  return {
    label,
    formatter,
    query: `
      // TODO: implement me
    `,
  };
}

export class AzureCosmosDataAdapterScripts extends BaseDataScript{
  // TODO: implement me
  getTableScripts() {
    return [
          getDivider,
    getSelectAllColumns,
    ]
  }

  // TODO: implement me
  getDatabaseScripts() {
    return [
    getDivider,
    getCreateDatabase,
    ]
  }

  // TODO: implement me
  getConnectionScripts() {
    return [
    ]
  }

  // TODO: implement me
  getSampleConnectionString(dialect?: SqluiCore.Dialect) {
    return `your_dialect://your_props`
  }
}
