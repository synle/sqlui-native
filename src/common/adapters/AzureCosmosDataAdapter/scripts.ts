import { getDivider } from 'src/common/adapters/BaseDataAdapter/scripts';
import { SqlAction, SqluiCore } from 'typings';

export const COSMOSDB_ADAPTER_PREFIX = 'db';

const COSMOSDB_TABLE_ALIAS_PREFIX = 'c';

const formatter = 'js';

export function getSampleConnectionString(dialect?: SqluiCore.Dialect) {
  return `cosmosdb://AccountEndpoint=some_cosmos_endpoint;AccountKey=some_cosmos_account_key`;
}

// https://docs.microsoft.com/en-us/azure/cosmos-db/sql/sql-api-nodejs-get-started?tabs=windows
function _shouldIncludeField(col: SqluiCore.ColumnMetaData) {
  if (col.name.indexOf('_') !== 0) {
    return true;
  }

  return false;
}

function _getColMapForInsertAndUpdate(columns?: SqluiCore.ColumnMetaData[]) {
  return (
    columns?.reduce((res, col) => {
      if (_shouldIncludeField(col)) {
        switch (col.type) {
          case 'string':
            res[col.name] = '';
            break;
          case 'number':
            res[col.name] = 123;
            break;
          case 'array':
            res[col.name] = [];
            break;
        }
      }

      return res;
    }, {}) || {}
  );
}

// for cosmosdb
export function getRawSelectAllColumns(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Raw Select All Columns SQL`;

  return {
    label,
    formatter: 'sql',
    query: `SELECT * FROM c`,
  };
}

export function getSelectAllColumns(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Select All Columns`;

  const sql = `
  SELECT *
  FROM ${COSMOSDB_TABLE_ALIAS_PREFIX}
  OFFSET 0 LIMIT ${input.querySize}`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .items
        .query({
          query: \`${sql}\`,
        })
        .fetchAll()
    `,
  };
}
export function getSelectById(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Select By Id`;

  const sql = `
  SELECT *
  FROM ${COSMOSDB_TABLE_ALIAS_PREFIX}
  WHERE ${COSMOSDB_TABLE_ALIAS_PREFIX}.id = '123'`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .items
        .query({
          query: \`${sql}\`,
        })
        .fetchAll()
    `,
  };
}

export function getReadItemById(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Read`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .item('some_id','some_partition_key')
        .read()
    `,
  };
}

export function getSelectSpecificColumns(
  input: SqlAction.TableInput,
): SqlAction.Output | undefined {
  const label = `Select Specific Columns`;

  const columnString = input?.columns
    ?.map((col) => `${COSMOSDB_TABLE_ALIAS_PREFIX}.${col.name}`)
    .join(',\n  ');
  const whereColumnString = input?.columns
    ?.map((col) => `${COSMOSDB_TABLE_ALIAS_PREFIX}.${col.name} = ''`)
    .join('\n  AND ');

  const sql = `
  SELECT ${columnString}
  FROM ${COSMOSDB_TABLE_ALIAS_PREFIX}
  WHERE ${whereColumnString}
  OFFSET 0 LIMIT ${input.querySize}`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .items
        .query({
          query: \`${sql}\`,
        })
        .fetchAll()
    `,
  };
}

export function getInsert(
  input: SqlAction.TableInput,
  value?: Record<string, any>,
): SqlAction.Output | undefined {
  const label = `Insert`;

  let colMap: any = {};
  if (value) {
    for (const key of Object.keys(value)) {
      colMap[key] = value[key];
    }
  } else {
    colMap = _getColMapForInsertAndUpdate(input?.columns);
  }

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .items
        .create(${JSON.stringify(colMap)})
    `,
  };
}

export function getUpdateWithValues(
  input: SqlAction.TableInput,
  value: Record<string, any>,
  conditions: Record<string, any>,
): SqlAction.Output | undefined {
  const label = `Update`;

  const colMap = _getColMapForInsertAndUpdate(input?.columns);

  if (value) {
    for (const key of Object.keys(value)) {
      colMap[key] = value[key];
    }
  }

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .item('some_id','some_partition_key')
        .replace(${JSON.stringify(colMap)})
    `,
  };
}
export function getUpdate(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Update`;

  const colMap = _getColMapForInsertAndUpdate(input?.columns);
  colMap['id'] = 'some_id';

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .item('some_id','some_partition_key')
        .replace(${JSON.stringify(colMap)})
    `,
  };
}

export function getDelete(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Delete`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .item('some_id','some_partition_key')
        .delete()
    `,
  };
}
export function getCreateContainer(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Create Container`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .containers
        .create({id: '${input.tableId}'})
    `,
  };
}

export function getDropContainer(input: SqlAction.TableInput): SqlAction.Output | undefined {
  const label = `Drop Container`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .container('${input.tableId}')
        .delete()
    `,
  };
}

export function getCreateDatabase(input: SqlAction.DatabaseInput): SqlAction.Output | undefined {
  const label = `Create Database`;

  return {
    label,
    formatter,
    query: `
      client
        .databases
        .create({id: '${input.databaseId}'})
    `,
  };
}

export function getCreateDatabaseContainer(
  input: SqlAction.DatabaseInput,
): SqlAction.Output | undefined {
  const label = `Create Database Container`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .containers
        .create({id: 'some_container_name'})
    `,
  };
}

export function getDropDatabase(input: SqlAction.DatabaseInput): SqlAction.Output | undefined {
  const label = `Drop Database`;

  return {
    label,
    formatter,
    query: `
      client
        .database('${input.databaseId}')
        .delete()
    `,
  };
}
export function getCreateConnectionDatabase(
  input: SqlAction.ConnectionInput,
): SqlAction.Output | undefined {
  const label = `Create Database`;

  return {
    label,
    formatter,
    query: `
      client
        .databases
        .create({id: 'some_database_name'})
    `,
  };
}

export const tableActionScripts: SqlAction.TableActionScriptGenerator[] = [
  getSelectAllColumns,
  getSelectSpecificColumns,
  getSelectById,
  getDivider,
  getReadItemById,
  getInsert,
  getUpdate,
  getDelete,
  getDivider,
  getRawSelectAllColumns,
  getDivider,
  getCreateContainer,
  getDropContainer,
];

export const databaseActionScripts: SqlAction.DatabaseActionScriptGenerator[] = [
  getDivider,
  getCreateDatabase,
  getCreateDatabaseContainer,
  getDivider,
  getDropDatabase,
];

export const connectionActionScripts: SqlAction.ConnectionActionScriptGenerator[] = [
  getDivider,
  getCreateConnectionDatabase,
];
