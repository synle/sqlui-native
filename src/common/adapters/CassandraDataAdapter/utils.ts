import BaseDataAdapter from 'src/common/adapters/BaseDataAdapter/index';

type ClientOptions = {
  host: string;
  port: number;
  contactPoints: string[];
  keyspace?: string;
  authProvider?: {
    username: string;
    password: string;
  };
};

export function getClientOptions(connectionOption: string, database?: string): ClientOptions {
  const connectionParameters = BaseDataAdapter.getConnectionParameters(connectionOption);

  const connectionHosts = connectionParameters?.hosts || [];
  if (connectionHosts.length === 0) {
    // we need a host in the connection string
    throw 'Invalid connection. Host and Port not found';
  }

  const host = connectionHosts[0].host;
  const port = connectionHosts[0].port || 9042;

  const clientOptions: ClientOptions = {
    host,
    port,
    contactPoints: [`${host}:${port}`],
  };

  if (database) {
    clientOptions.keyspace = database;
  }

  // client authentication
  if (connectionParameters?.username && connectionParameters?.password) {
    clientOptions.authProvider = {
      username: connectionParameters?.username,
      password: connectionParameters?.password,
    };
  }

  return clientOptions;
}