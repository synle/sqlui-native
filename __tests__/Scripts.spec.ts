import fs from 'fs';
import { SqluiCore, SqlAction } from 'typings';
import { formatJS } from 'src/utils/formatter';
import { formatSQL } from 'src/utils/formatter';
import { scripts as RmdbScripts } from 'src/scripts/rmdb';
import { scripts as CassandraScripts } from 'src/scripts/cassandra';
import { scripts as MongodbScripts } from 'src/scripts/mongodb';
import { scripts as RedisScripts } from 'src/scripts/redis';

function _getScript(scripts: SqlAction.ScriptGenerator[], dialect: SqluiCore.Dialect) {
  return scripts
    .map((fn) => {
      return fn({
        dialect,
        connectionId: 'connection1',
        databaseId: 'database1',
        querySize: 200,
        tableId: 'table1',
        columns: [
          {
            name: 'column1',
            type: 'NUMBER',
          },
          {
            name: 'column2',
            type: 'VARCHAR(100)',
          },
        ],
      });
    })
    .filter((script) => script);
}

describe('Scripts', () => {
  let commandGuides: string[] = [`
---
title: sqlui-native query help?
---

Query Help?
===========
  `.trim()];

  function addGuideText(sectionName: string, scripts: (SqlAction.Output | undefined)[]) {
    commandGuides.push(`## ${sectionName}\n`);

    for (const script of scripts) {
      if (script && script.query) {
        commandGuides.push(`### ${script.label}\n`);

        let query = script.query
          .replace(/--/g, '\n')
          .replace(/\n/g, ' ')
          .replace(/[ ][ ]+/g, ' ')
          .trim();
        switch (script.formatter) {
          case 'sql':
            query = formatSQL(query);
            break;
          case 'js':
            query = formatJS(query);
            break;
        }

        commandGuides.push('```');
        commandGuides.push(query);
        commandGuides.push('```\n\n');
      }
    }
  }

  test('RmdbScripts - mysql', async () => {
    const scripts = _getScript(RmdbScripts, 'mysql');
    expect(scripts).toMatchSnapshot();
    addGuideText('mysql', scripts);
  });

  test('RmdbScripts - mariadb', async () => {
    const scripts = _getScript(RmdbScripts, 'mariadb');
    expect(scripts).toMatchSnapshot();
    addGuideText('mariadb', scripts);
  });

  test('RmdbScripts - mssql', async () => {
    const scripts = _getScript(RmdbScripts, 'mssql');
    expect(scripts).toMatchSnapshot();
    addGuideText('mssql', scripts);
  });

  test('RmdbScripts - postgres', async () => {
    const scripts = _getScript(RmdbScripts, 'postgres');
    expect(scripts).toMatchSnapshot();
    addGuideText('postgres', scripts);
  });

  test('RmdbScripts - sqlite', async () => {
    const scripts = _getScript(RmdbScripts, 'sqlite');
    expect(scripts).toMatchSnapshot();
    addGuideText('sqlite', scripts);
  });

  test('CassandraScripts', async () => {
    const scripts = _getScript(CassandraScripts, 'cassandra');
    expect(scripts).toMatchSnapshot();
    addGuideText('cassandra', scripts);
  });

  test('MongodbScripts', async () => {
    const scripts = _getScript(MongodbScripts, 'mongodb');
    expect(scripts).toMatchSnapshot();
    addGuideText('mongodb', scripts);
  });

  test('RedisScripts', async () => {
    const scripts = _getScript(RedisScripts, 'redis');
    expect(scripts).toMatchSnapshot();
    addGuideText('redis', scripts);
  });

  test('Consolidate the guide into a command', async () => {
    const newGuide = commandGuides.join('\n');
    fs.writeFileSync('./guides.md', newGuide);
  });
});
