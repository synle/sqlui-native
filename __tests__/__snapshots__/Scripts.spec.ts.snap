// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scripts CassandraScripts 1`] = `
Array [
  Object {
    "formatter": "sql",
    "label": "Select All Columns",
    "query": "SELECT * 
FROM table1 
LIMIT 200",
  },
  Object {
    "formatter": "sql",
    "label": "Select Specific Columns",
    "query": "SELECT 
  column1,
  column2 
FROM table1 
 -- WHERE column1 = ''
 -- AND column2 = '' 
LIMIT 200",
  },
  Object {
    "formatter": "sql",
    "label": "Insert",
    "query": "INSERT INTO table1 (
column1,
column2
) VALUES (
'_column1_',
'_column2_'
)",
  },
  Object {
    "formatter": "js",
    "label": "Update",
    "query": "UPDATE table1
 SET 
-- column1 = '',
-- column2 = ''
 WHERE -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Delete",
    "query": "-- DELETE FROM table1 
 -- WHERE
 -- column1 = '' AND 
-- column2 = ''",
  },
]
`;

exports[`Scripts MongodbScripts 1`] = `
Array [
  Object {
    "formatter": "sql",
    "label": "Select All Columns",
    "query": "db.collection('table1').find().limit(200).toArray();",
  },
  Object {
    "formatter": "sql",
    "label": "Select Specific Columns",
    "query": "db.collection('table1').find(
          {\\"column1\\":123,\\"column2\\":123}
        ).limit(200).toArray();",
  },
  Object {
    "formatter": "js",
    "label": "Insert",
    "query": "db.collection('table1').insertMany([
          {\\"column1\\":123,\\"column2\\":123}
        ]);",
  },
  Object {
    "formatter": "js",
    "label": "Update",
    "query": "db.collection('table1').update(
          {\\"column1\\":123,\\"column2\\":123},
          {$set: {
  \\"column1\\": 123,
  \\"column2\\": 123
}}
        );",
  },
  Object {
    "formatter": "js",
    "label": "Delete",
    "query": "db.collection('table1').deleteMany(
          {\\"column1\\":123,\\"column2\\":123}
        );",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "js",
    "label": "Create Table",
    "query": "db.createCollection(\\"table1\\")",
  },
  Object {
    "formatter": "js",
    "label": "Drop Table",
    "query": "db.collection('table1').drop()",
  },
]
`;

exports[`Scripts RedisScripts 1`] = `
Array [
  Object {
    "formatter": "js",
    "label": "Set Value",
    "query": "db.set(\\"key\\", \\"value123\\")",
  },
  Object {
    "formatter": "js",
    "label": "Get Value by Key",
    "query": "db.get(\\"key\\")",
  },
  Object {
    "formatter": "js",
    "label": "Scan for keys",
    "query": "db.keys(\\"*\\")",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "js",
    "label": "Hashset > Set Value",
    "query": "db.hSet(\\"key\\", \\"field\\", \\"value\\")",
  },
  Object {
    "formatter": "js",
    "label": "Hashset > Get Value By Key",
    "query": "db.hGetAll(\\"key\\")",
  },
  Object {
    "formatter": "js",
    "label": "Hashset > Values",
    "query": "db.hVals(\\"key\\")",
  },
  Object {
    "formatter": "js",
    "label": "Hashset > Check if key exists",
    "query": "db.hExists(\\"key\\", \\"field1\\")",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "js",
    "label": "List > Get Items",
    "query": "db.lRange(\\"key\\", 0, -1)",
  },
  Object {
    "formatter": "js",
    "label": "List > Push item to the front",
    "query": "db.lPush(\\"key\\", \\"value\\")",
  },
  Object {
    "formatter": "js",
    "label": "List > Push item to the back",
    "query": "db.rPush(\\"key\\", \\"value\\")",
  },
  Object {
    "formatter": "js",
    "label": "List > Delete item from the front",
    "query": "db.lPop(\\"key\\")",
  },
  Object {
    "formatter": "js",
    "label": "List > Delete item from the back",
    "query": "db.rPop(\\"key\\")",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "js",
    "label": "Set > Get Items",
    "query": "db.sMembers(\\"key\\")",
  },
  Object {
    "formatter": "js",
    "label": "Set > Add Item",
    "query": "db.sAdd(\\"key\\", \\"value1\\")",
  },
  Object {
    "formatter": "js",
    "label": "Set > Is a member of set",
    "query": "db.sIsMember(\\"key\\", \\"value1\\")",
  },
  Object {
    "formatter": "js",
    "label": "Set > Total Size",
    "query": "db.sCard(\\"key\\")",
  },
  Object {
    "formatter": "js",
    "label": "Set > Remove last item",
    "query": "db.sPop(\\"key\\")",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "js",
    "label": "Sorted Set > Get Items",
    "query": "db.zRange(\\"key\\", 0, -1)",
  },
  Object {
    "formatter": "js",
    "label": "Sorted Set > Add Item",
    "query": "db.zAdd(\\"key\\", [{score: 1, value: \\"some_value\\"}])",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "js",
    "label": "Publish a message",
    "query": "db.publish(\\"key\\", \\"some_message\\")",
  },
]
`;

exports[`Scripts RmdbScripts - mariadb 1`] = `
Array [
  Object {
    "formatter": "sql",
    "label": "Select All Columns",
    "query": "SELECT * 
FROM table1 
LIMIT 200",
  },
  Object {
    "formatter": "sql",
    "label": "Select Count",
    "query": "SELECT COUNT(*) 
FROM table1 
 -- WHERE 
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Select Specific Columns",
    "query": "SELECT 
  column1,
  column2 
FROM table1 
 -- WHERE column1 = ''
 -- AND column2 = '' 
LIMIT 200",
  },
  Object {
    "formatter": "js",
    "label": "Update",
    "query": "UPDATE table1
 SET 
-- column1 = '',
-- column2 = ''
 WHERE -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Delete",
    "query": "-- DELETE FROM table1 
 -- WHERE
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "sql",
    "label": "Create Table",
    "query": "CREATE TABLE table1 (column1 NUMBER   NOT NULL,
column2 VARCHAR(100)   NOT NULL)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Table",
    "query": "-- DROP TABLE table1",
  },
  Object {
    "formatter": "sql",
    "label": "Add Column",
    "query": "ALTER TABLE table1 ADD COLUMN newColumn1 varchar(200)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Column",
    "query": "--ALTER TABLE table1 DROP COLUMN column1
--ALTER TABLE table1 DROP COLUMN column2",
  },
]
`;

exports[`Scripts RmdbScripts - mssql 1`] = `
Array [
  Object {
    "formatter": "sql",
    "label": "Select All Columns",
    "query": "SELECT TOP 200 * 
FROM table1",
  },
  Object {
    "formatter": "sql",
    "label": "Select Count",
    "query": "SELECT COUNT(*) 
FROM table1 
 -- WHERE 
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Select Specific Columns",
    "query": "SELECT TOP 200 
  column1,
  column2 
FROM table1 -- WHERE column1 = ''
 -- AND column2 = ''",
  },
  Object {
    "formatter": "js",
    "label": "Update",
    "query": "UPDATE table1
 SET 
-- column1 = '',
-- column2 = ''
 WHERE -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Delete",
    "query": "-- DELETE FROM table1 
 -- WHERE
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "sql",
    "label": "Create Table",
    "query": "CREATE TABLE table1 (column1 NUMBER   NOT NULL,
column2 VARCHAR(100)   NOT NULL)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Table",
    "query": "-- DROP TABLE table1",
  },
  Object {
    "formatter": "sql",
    "label": "Add Column",
    "query": "ALTER TABLE table1 ADD COLUMN newColumn1 NVARCHAR(200)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Column",
    "query": "--ALTER TABLE table1 DROP COLUMN column1
--ALTER TABLE table1 DROP COLUMN column2",
  },
]
`;

exports[`Scripts RmdbScripts - mysql 1`] = `
Array [
  Object {
    "formatter": "sql",
    "label": "Select All Columns",
    "query": "SELECT * 
FROM table1 
LIMIT 200",
  },
  Object {
    "formatter": "sql",
    "label": "Select Count",
    "query": "SELECT COUNT(*) 
FROM table1 
 -- WHERE 
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Select Specific Columns",
    "query": "SELECT 
  column1,
  column2 
FROM table1 
 -- WHERE column1 = ''
 -- AND column2 = '' 
LIMIT 200",
  },
  Object {
    "formatter": "js",
    "label": "Update",
    "query": "UPDATE table1
 SET 
-- column1 = '',
-- column2 = ''
 WHERE -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Delete",
    "query": "-- DELETE FROM table1 
 -- WHERE
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "sql",
    "label": "Create Table",
    "query": "CREATE TABLE table1 (column1 NUMBER   NOT NULL,
column2 VARCHAR(100)   NOT NULL)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Table",
    "query": "-- DROP TABLE table1",
  },
  Object {
    "formatter": "sql",
    "label": "Add Column",
    "query": "ALTER TABLE table1 ADD COLUMN newColumn1 varchar(200)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Column",
    "query": "--ALTER TABLE table1 DROP COLUMN column1
--ALTER TABLE table1 DROP COLUMN column2",
  },
]
`;

exports[`Scripts RmdbScripts - postgres 1`] = `
Array [
  Object {
    "formatter": "sql",
    "label": "Select All Columns",
    "query": "SELECT * 
FROM table1 
LIMIT 200",
  },
  Object {
    "formatter": "sql",
    "label": "Select Count",
    "query": "SELECT COUNT(*) 
FROM table1 
 -- WHERE 
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Select Specific Columns",
    "query": "SELECT 
  column1,
  column2 
FROM table1 
 -- WHERE column1 = ''
 -- AND column2 = '' 
LIMIT 200",
  },
  Object {
    "formatter": "js",
    "label": "Update",
    "query": "UPDATE table1
 SET 
-- column1 = '',
-- column2 = ''
 WHERE -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Delete",
    "query": "-- DELETE FROM table1 
 -- WHERE
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "sql",
    "label": "Create Table",
    "query": "CREATE TABLE table1 (column1 NUMBER NOT NULL,
column2 VARCHAR(100) NOT NULL)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Table",
    "query": "-- DROP TABLE table1",
  },
  Object {
    "formatter": "sql",
    "label": "Add Column",
    "query": "ALTER TABLE table1 ADD COLUMN newColumn1 CHAR(200)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Column",
    "query": "--ALTER TABLE table1 DROP COLUMN column1
--ALTER TABLE table1 DROP COLUMN column2",
  },
]
`;

exports[`Scripts RmdbScripts - sqlite 1`] = `
Array [
  Object {
    "formatter": "sql",
    "label": "Select All Columns",
    "query": "SELECT * 
FROM table1 
LIMIT 200",
  },
  Object {
    "formatter": "sql",
    "label": "Select Count",
    "query": "SELECT COUNT(*) 
FROM table1 
 -- WHERE 
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Select Specific Columns",
    "query": "SELECT 
  column1,
  column2 
FROM table1 
 -- WHERE column1 = ''
 -- AND column2 = '' 
LIMIT 200",
  },
  Object {
    "formatter": "js",
    "label": "Update",
    "query": "UPDATE table1
 SET 
-- column1 = '',
-- column2 = ''
 WHERE -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "formatter": "sql",
    "label": "Delete",
    "query": "-- DELETE FROM table1 
 -- WHERE
 -- column1 = '' AND 
-- column2 = ''",
  },
  Object {
    "label": "divider",
  },
  Object {
    "formatter": "sql",
    "label": "Create Table",
    "query": "CREATE TABLE table1 (column1 NUMBER   NOT NULL,
column2 VARCHAR(100)   NOT NULL)",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Table",
    "query": "-- DROP TABLE table1",
  },
  Object {
    "formatter": "sql",
    "label": "Add Column",
    "query": "ALTER TABLE table1 ADD COLUMN newColumn1 TEXT",
  },
  Object {
    "formatter": "sql",
    "label": "Drop Column",
    "query": "--ALTER TABLE table1 DROP COLUMN column1
--ALTER TABLE table1 DROP COLUMN column2",
  },
]
`;
